- PassegerId: El código de identificación del pasajero (valor único).
- Survived: Variable factor, 1 (el pasajero sobrevivió) 0 (el pasajero no sobrevivió).
- Pclass: En que clase viajaba el pasajero (1 = primera, 2 = segunda, 3 = tercera).
- Name: Nombre del pasajero (valor único).
- Sex: Sexo del pasajero.
- Age: Edad del pasajero.
- SibSp: Cantidad de hermanos o cónyuges a bordo del - Titanic.
- Parch: Cantidad de padres o hijos a bordo del Titanic.
- Ticket: Número de tiquete (valor único).
- Fare: Tarifa del pasajero.
- Cabin: Número de cabina (valor único).
- Embarked: Puerto donde embarco el pasajero (C = Cherbourg, Q = Queenstown, S = Southampton).

In [4]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt


In [5]:
df = pd.read_csv("full.csv")

pd.set_option('display.max_columns', None)

In [6]:
df.head(20)

PassengerId  Survived  Pclass  \
0             1       0.0       3   
1             2       1.0       1   
2             3       1.0       3   
3             4       1.0       1   
4             5       0.0       3   
5             6       0.0       3   
6             7       0.0       1   
7             8       0.0       3   
8             9       1.0       3   
9            10       1.0       2   
10           11       1.0       3   
11           12       1.0       1   
12           13       0.0       3   
13           14       0.0       3   
14           15       0.0       3   
15           16       1.0       2   
16           17       0.0       3   
17           18       1.0       2   
18           19       0.0       3   
19           20       1.0       3   

                                                 Name     Sex   Age  SibSp  \
0                             Braund, Mr. Owen Harris    male  22.0      1   
1   Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                              Heikkinen, Miss. Laina  female  26.0      0   
3        Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                            Allen, Mr. William Henry    male  35.0      0   
5                                    Moran, Mr. James    male   NaN      0   
6                             McCarthy, Mr. Timothy J    male  54.0      0   
7                      Palsson, Master. Gosta Leonard    male   2.0      3   
8   Johnson, Mrs. Oscar W (Elisabeth Vilhelmina Berg)  female  27.0      0   
9                 Nasser, Mrs. Nicholas (Adele Achem)  female  14.0      1   
10                    Sandstrom, Miss. Marguerite Rut  female   4.0      1   
11                           Bonnell, Miss. Elizabeth  female  58.0      0   
12                     Saundercock, Mr. William Henry    male  20.0      0   
13                        Andersson, Mr. Anders Johan    male  39.0      1   
14               Vestrom, Miss. Hulda Amanda Adolfina  female  14.0      0   
15                   Hewlett, Mrs. (Mary D Kingcome)   female  55.0      0   
16                               Rice, Master. Eugene    male   2.0      4   
17                       Williams, Mr. Charles Eugene    male   NaN      0   
18  Vander Planke, Mrs. Julius (Emelia Maria Vande...  female  31.0      1   
19                            Masselmani, Mrs. Fatima  female   NaN      0   

    Parch            Ticket     Fare Cabin Embarked  WikiId  \
0       0         A/5 21171   7.2500   NaN        S   691.0   
1       0          PC 17599  71.2833   C85        C    90.0   
2       0  STON/O2. 3101282   7.9250   NaN        S   865.0   
3       0            113803  53.1000  C123        S   127.0   
4       0            373450   8.0500   NaN        S   627.0   
5       0            330877   8.4583   NaN        Q   785.0   
6       0             17463  51.8625   E46        S   200.0   
7       1            349909  21.0750   NaN        S  1108.0   
8       2            347742  11.1333   NaN        S   902.0   
9       0            237736  30.0708   NaN        C   520.0   
10      1           PP 9549  16.7000    G6        S  1193.0   
11      0            113783  26.5500  C103        S    35.0   
12      0         A/5. 2151   8.0500   NaN        S  1196.0   
13      5            347082  31.2750   NaN        S   632.0   
14      0            350406   7.8542   NaN        S  1286.0   
15      0            248706  16.0000   NaN        S   446.0   
16      1            382652  29.1250   NaN        Q  1151.0   
17      0            244373  13.0000   NaN        S   604.0   
18      0            345763  18.0000   NaN        S  1280.0   
19      0              2649   7.2250   NaN        C  1053.0   

                                            Name_wiki  Age_wiki  \
0                             Braund, Mr. Owen Harris      22.0   
1          Cumings, Mrs. Florence Briggs (née Thayer)      35.0   
2                               Heikkinen, Miss Laina      26.

In [7]:
# Muestra los valores totales del DataFrame
df.size
print(f'El DataFrame tiene {df.size} valores')

El DataFrame tiene 27489 valores


In [8]:
print(f"El número de filas que tenemos es {df.shape[0]}, y el número de columnas es {df.shape[1]}")

El número de filas que tenemos es 1309, y el número de columnas es 21


In [9]:
df.columns

Index(['PassengerId', 'Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp',
       'Parch', 'Ticket', 'Fare', 'Cabin', 'Embarked', 'WikiId', 'Name_wiki',
       'Age_wiki', 'Hometown', 'Boarded', 'Destination', 'Lifeboat', 'Body',
       'Class'],
      dtype='object')

In [10]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1309 entries, 0 to 1308
Data columns (total 21 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  1309 non-null   int64  
 1   Survived     891 non-null    float64
 2   Pclass       1309 non-null   int64  
 3   Name         1309 non-null   object 
 4   Sex          1309 non-null   object 
 5   Age          1046 non-null   float64
 6   SibSp        1309 non-null   int64  
 7   Parch        1309 non-null   int64  
 8   Ticket       1309 non-null   object 
 9   Fare         1308 non-null   float64
 10  Cabin        295 non-null    object 
 11  Embarked     1307 non-null   object 
 12  WikiId       1304 non-null   float64
 13  Name_wiki    1304 non-null   object 
 14  Age_wiki     1302 non-null   float64
 15  Hometown     1304 non-null   object 
 16  Boarded      1304 non-null   object 
 17  Destination  1304 non-null   object 
 18  Lifeboat     502 non-null    object 
 19  Body  

In [11]:
df.dtypes

PassengerId      int64
Survived       float64
Pclass           int64
Name            object
Sex             object
Age            float64
SibSp            int64
Parch            int64
Ticket          object
Fare           float64
Cabin           object
Embarked        object
WikiId         float64
Name_wiki       object
Age_wiki       float64
Hometown        object
Boarded         object
Destination     object
Lifeboat        object
Body            object
Class          float64
dtype: object

In [12]:
df.describe()    

PassengerId    Survived       Pclass          Age        SibSp  \
count  1309.000000  891.000000  1309.000000  1046.000000  1309.000000   
mean    655.000000    0.383838     2.294882    29.881138     0.498854   
std     378.020061    0.486592     0.837836    14.413493     1.041658   
min       1.000000    0.000000     1.000000     0.170000     0.000000   
25%     328.000000    0.000000     2.000000    21.000000     0.000000   
50%     655.000000    0.000000     3.000000    28.000000     0.000000   
75%     982.000000    1.000000     3.000000    39.000000     1.000000   
max    1309.000000    1.000000     3.000000    80.000000     8.000000   

             Parch         Fare       WikiId     Age_wiki        Class  
count  1309.000000  1308.000000  1304.000000  1302.000000  1304.000000  
mean      0.385027    33.295479   658.534509    29.415829     2.291411  
std       0.865560    51.758668   380.377373    13.758954     0.840852  
min       0.000000     0.000000     1.000000     0.170000     1.000000  
25%       0.000000     7.895800   326.750000    21.000000     1.750000  
50%       0.000000    14.454200   661.500000    28.000000     3.000000  
75%       0.000000    31.275000   987.250000    37.750000     3.000000  
max       9.000000   512.329200  1314.000000    74.000000     3.000000

Nota : Solo tenomos una columna numérica.
Podemos buscar otro archivo que nos valga y unirlo.

In [13]:
# Ver un resumen para variables CATEGÓRICAS
df.describe(include=['object']).T

count unique                     top freq
Name         1309   1307    Connolly, Miss. Kate    2
Sex          1309      2                    male  843
Ticket       1309    929                CA. 2343   11
Cabin         295    186             C23 C25 C27    6
Embarked     1307      3                       S  914
Name_wiki    1304   1303        Kelly, Mr. James    2
Hometown     1304    566  New York, New York, US   70
Boarded      1304      4             Southampton  916
Destination  1304    291           New York City  234
Lifeboat      502     24                      13   42
Body          130    130                   175MB    1

In [14]:
# Contar frecuencias de variables categóricas
for column in df.select_dtypes(include='object').columns:
    print(f"\nFrecuencias en la columna '{column}':")
    print(f"Los valores únicos son: {df[column].unique()}\n")
    print(f"Las frecuncias de los valores unicos de las categorías son: {df[column].value_counts()}")


Frecuencias en la columna 'Name':
Los valores únicos son: ['Braund, Mr. Owen Harris'
 'Cumings, Mrs. John Bradley (Florence Briggs Thayer)'
 'Heikkinen, Miss. Laina' ... 'Saether, Mr. Simon Sivertsen'
 'Ware, Mr. Frederick' 'Peter, Master. Michael J']

Las frecuncias de los valores unicos de las categorías son: Name
Connolly, Miss. Kate                    2
Kelly, Mr. James                        2
Vestrom, Miss. Hulda Amanda Adolfina    1
Hewlett, Mrs. (Mary D Kingcome)         1
Rice, Master. Eugene                    1
                                       ..
Foley, Mr. Joseph                       1
Risien, Mrs. Samuel (Emma)              1
McNamee, Mrs. Neal (Eileen O'Leary)     1
Wheeler, Mr. Edwin Frederick""          1
Saundercock, Mr. William Henry          1
Name: count, Length: 1307, dtype: int64

Frecuencias en la columna 'Sex':
Los valores únicos son: ['male' 'female']

Las frecuncias de los valores unicos de las categorías son: Sex
male      843
female    466
Name: coun

In [15]:
df.isnull().sum()

PassengerId       0
Survived        418
Pclass            0
Name              0
Sex               0
Age             263
SibSp             0
Parch             0
Ticket            0
Fare              1
Cabin          1014
Embarked          2
WikiId            5
Name_wiki         5
Age_wiki          7
Hometown          5
Boarded           5
Destination       5
Lifeboat        807
Body           1179
Class             5
dtype: int64

In [16]:
# Mostrar solo las columnas con los valores nulos y el total de ellos
df.isna().sum()[df.isna().sum() > 0]

Survived        418
Age             263
Fare              1
Cabin          1014
Embarked          2
WikiId            5
Name_wiki         5
Age_wiki          7
Hometown          5
Boarded           5
Destination       5
Lifeboat        807
Body           1179
Class             5
dtype: int64

In [17]:
#Top 10 países con más contenido
countries = df['Hometown'].value_counts()
countries

Hometown
New York, New York, US            70
London, England                   49
London, England, UK               30
Paris, France                     23
Montreal, Quebec, Canada          20
                                  ..
Maidenhead, Berkshire, England     1
Torquay, Devon, England, UK        1
Belmont, Massachusetts, US         1
Sandhurst, Kent, England           1
Waterford, Waterford, Ireland      1
Name: count, Length: 566, dtype: int64

In [18]:
# Contar valores duplicados en todo el dataset
duplicados = df.duplicated().sum()
print(f"Valores duplicados en el dataset: {duplicados}")

Valores duplicados en el dataset: 0


In [19]:
# Calculamos los valores faltantes por columna
missing_values = df.isnull().sum()

# Calculamos el porcentaje de valores faltantes
missing_percentage = (missing_values / len(df)) * 100

In [ ]:
# Creamos un DataFrame de valores faltantes
missing_values_df = pd.DataFrame({
    'Column': missing_values.index,
    'Missing Values': missing_values.values,
    'Percentage': missing_percentage
}).sort_values(by='Missing Values', ascending=False)

missing_values_df.head(6)

Column  Missing Values  Percentage
Body          Body            1179   90.068755
Cabin        Cabin            1014   77.463713
Lifeboat  Lifeboat             807   61.650115
Survived  Survived             418   31.932773
Age            Age             263   20.091673
Age_wiki  Age_wiki               7    0.534759

In [32]:
# Comprobamos los valores de las clases

df['Pclass'].value_counts()

Pclass
3    709
1    323
2    277
Name: count, dtype: int64

In [21]:
titanic = sns.load_dataset("titanic")
titanic.head()

survived  pclass     sex   age  sibsp  parch     fare embarked  class  \
0         0       3    male  22.0      1      0   7.2500        S  Third   
1         1       1  female  38.0      1      0  71.2833        C  First   
2         1       3  female  26.0      0      0   7.9250        S  Third   
3         1       1  female  35.0      1      0  53.1000        S  First   
4         0       3    male  35.0      0      0   8.0500        S  Third   

     who  adult_male deck  embark_town alive  alone  
0    man        True  NaN  Southampton    no  False  
1  woman       False    C    Cherbourg   yes  False  
2  woman       False  NaN  Southampton   yes   True  
3  woman       False    C  Southampton   yes  False  
4    man        True  NaN  Southampton    no   True

In [22]:
print(f"El número de filas que tenemos es {titanic.shape[0]}, y el número de columnas es {titanic.shape[1]}")

El número de filas que tenemos es 891, y el número de columnas es 15


In [23]:
# Mostrar solo las columnas con los valores nulos y el total de ellos
titanic.isna().sum()[titanic.isna().sum() > 0]

age            177
embarked         2
deck           688
embark_town      2
dtype: int64

In [24]:
df.isna().sum()[df.isna().sum() > 0]

Survived        418
Age             263
Fare              1
Cabin          1014
Embarked          2
WikiId            5
Name_wiki         5
Age_wiki          7
Hometown          5
Boarded           5
Destination       5
Lifeboat        807
Body           1179
Class             5
dtype: int64

In [25]:
titanic.isna().sum()[titanic.isna().sum() > 0]

age            177
embarked         2
deck           688
embark_town      2
dtype: int64

In [26]:
# Rellenar la columna 'A' de df1 con los valores de la columna 'X' de df2
df['Age'] = titanic['age']

In [27]:
df.isna().sum()[df.isna().sum() > 0]

Survived        418
Age             595
Fare              1
Cabin          1014
Embarked          2
WikiId            5
Name_wiki         5
Age_wiki          7
Hometown          5
Boarded           5
Destination       5
Lifeboat        807
Body           1179
Class             5
dtype: int64

In [28]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1309 entries, 0 to 1308
Data columns (total 21 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  1309 non-null   int64  
 1   Survived     891 non-null    float64
 2   Pclass       1309 non-null   int64  
 3   Name         1309 non-null   object 
 4   Sex          1309 non-null   object 
 5   Age          714 non-null    float64
 6   SibSp        1309 non-null   int64  
 7   Parch        1309 non-null   int64  
 8   Ticket       1309 non-null   object 
 9   Fare         1308 non-null   float64
 10  Cabin        295 non-null    object 
 11  Embarked     1307 non-null   object 
 12  WikiId       1304 non-null   float64
 13  Name_wiki    1304 non-null   object 
 14  Age_wiki     1302 non-null   float64
 15  Hometown     1304 non-null   object 
 16  Boarded      1304 non-null   object 
 17  Destination  1304 non-null   object 
 18  Lifeboat     502 non-null    object 
 19  Body  

In [29]:
df['Destination'].value_counts()

Destination
New York City                  234
New York, New York, US         127
Chicago, Illinois, US           75
Montreal, Quebec, Canada        37
Detroit, Michigan, US           28
                              ... 
Gloversville, New York, US       1
Brighton, Massachusetts, US      1
Milwaukee, Wisconsin             1
Sherbrooke, Quebec, Canada       1
US                               1
Name: count, Length: 291, dtype: int64

In [30]:
df['Lifeboat'].value_counts()

Lifeboat
13       42
C        41
15       38
14       34
4        31
10       29
5        29
11       26
9        26
3        26
8        24
16       23
7        22
6        21
D        18
12       18
2        14
?        12
A        11
B         9
1         5
14?       1
A[64]     1
15?       1
Name: count, dtype: int64

In [31]:
df['Boarded'].value_counts()

Boarded
Southampton    916
Cherbourg      259
Queenstown     119
Belfast         10
Name: count, dtype: int64

# CONCLUSIONES
- Hay 14 columnas con valores nulos
- Survived --> 0 = no; 1 = sí y 418 valores nulos. Cambiar a si y no
- Eliminar: Age, Name_wiki, Body, WikiId, Lifeboat y Class
- Cambiar de float a int: Age_wiki, Survived
- Pasar la columna sex de titanic a df y cambiarle el nombre a gender
- Reemplazar en Destination New York City por New York, New York, US 
- Cambiar el nombre de algunas columnas: Pclass: Class y Age_wiki: Age
- Redondear la columna Fare a dos decimales
- Poner en las columnas categóricas con nulos 'Unknown'
- Cambiar Gender male = M y female = F
- Name: tiene Mr (M), Mrs o Miss (F) por que se va a predecir con esos tratamientos si es male (M) o female(F)
- Se completa Boarded con Embarked y se borra esta última